state/state.go

```diff
@@ -142,38 +148,43 @@ func (s *State) SaveABCIResponses(abciResponses *ABCIResponses) {
 }
 
 // LoadABCIResponses loads the ABCIResponses from the database.
+// This is useful for recovering from crashes where we called app.Commit and before we called
+// s.Save()
 func (s *State) LoadABCIResponses() *ABCIResponses {
-	abciResponses := new(ABCIResponses)
-
 	buf := s.db.Get(abciResponsesKey)
-	if len(buf) != 0 {
-		r, n, err := bytes.NewReader(buf), new(int), new(error)
-		wire.ReadBinaryPtr(abciResponses, r, 0, n, err)
-		if *err != nil {
-			// DATA HAS BEEN CORRUPTED OR THE SPEC HAS CHANGED
-			cmn.Exit(cmn.Fmt("LoadABCIResponses: Data has been corrupted or its spec has changed: %v\n", *err))
-		}
-		// TODO: ensure that buf is completely read.
+	if len(buf) == 0 {
+		return nil
```

Comment:
Unfortunately this might be an unintended change because, previously in the case that `len(buf) == 0` we always reeturned `new(ABCIResponses)` now this change returns `nil`. Hopefully tests can catch this behavior.

Reply:
It does, but currently at least nothing within Tendermint relies on this. However now it follows the semantics of `LoadState` in that if there is nothing it will return `nil`. I think we should change it now and broken clients hopefully catch it, because it seems strange to have different semantics in two cases which are very similar.

I've made a note in the changelog.